openmc/mgxs_library.py

```diff
@@ -603,7 +603,8 @@ def nu_fission(self, nu_fission):
         if np.sum(self._nu_fission) > 0.0:
             self._fissionable = True
 
-    def set_total_mgxs(self, total, nuclide='total', xs_type='macro'):
+    def set_total_mgxs(self, total, nuclide='total', xs_type='macro',
+                       subdomain='all'):
```

Comment:
Elsewhere this parameter is typically plural as `subdomains` since a user can specify a list of one, some or all subdomains (i.e., distribcell IDs or mesh cell indices), or the `'all'` string. In this case, it looks like you can either specify a single subdomain (mesh cell indices only) or `'all'`. Can this be expanded to be more flexible and accept one, some or all subdomains as is done elsewhere in `openmc.mgxs`?